### PR TITLE
Fix quiz detail when quiz submission doesn't load

### PR DESCRIPTION
**Pull request by vanwagonet**
refs: MBL-12538
affects: none
release note: none

### CODE REVIEW

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: None

[MBL-12538](https://instructure.atlassian.net/browse/MBL-12538)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Core** | 95.17% | 95.17% | 0%
**Student** | 92.9% | 92.89% | 0.01%
**Teacher** | 100% | 100% | 0%
**React Native** | 92.7% | 92.7% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Student/Student/Assignments/AssignmentDetails/AssignmentDetailsPresenter.swift [hunk 1]
```diff
@@ -84,6 +84,7 @@ class AssignmentDetailsPresenter {
             quizzes?.refresh()
         }
         guard let assignment = assignment, let course = courses.first else { return }
+        guard quizzes?.pending != true else { return }
```
Comment (vanwagonet) [inline Student/Student/Assignments/AssignmentDetails/AssignmentDetailsPresenter.swift]: wait for potential quiz to come back

File: Student/Student/Assignments/AssignmentDetails/AssignmentDetailsViewController.swift [hunk 2]
```diff
@@ -85,6 +85,7 @@ class AssignmentDetailsViewController: UIViewController, AssignmentDetailsViewPr
         refreshControl.addTarget(self, action: #selector(refresh(_:)), for: .valueChanged)
         scrollView?.addSubview(refreshControl)
         self.refreshControl = refreshControl
+        showSubmitAssignmentButton(title: nil)
```
Comment (vanwagonet) [inline Student/Student/Assignments/AssignmentDetails/AssignmentDetailsViewController.swift]: fix button flashing "Submit Assignment" before going to the correct string.

File: Student/Student/Submissions/SubmissionButton/SubmissionButtonPresenter.swift [hunk 3]
```diff
@@ -68,7 +68,7 @@ class SubmissionButtonPresenter: NSObject {
             return NSLocalizedString("Resume Quiz", bundle: .student, comment: "")
         }
         if quiz?.submission?.attemptsLeft == 0 { return nil }
-        if quiz != nil {
+        if assignment.quizID != nil {
```
Comment (vanwagonet) [inline Student/Student/Submissions/SubmissionButton/SubmissionButtonPresenter.swift]: if quiz failed to load, we still know it was supposed to be a quiz.

**Pull request merged**